gtsam/nonlinear/ISAM2Params.h

```diff
@@ -300,50 +300,24 @@ struct GTSAM_EXPORT ISAM2Params {
   RelinearizationThreshold getRelinearizeThreshold() const {
     return relinearizeThreshold;
   }
-  int getRelinearizeSkip() const { return relinearizeSkip; }
```

Comment:
Why are we removing all these methods? Isn't this API breaking and tangential to the point of this PR?

Reply:
> Why are we removing all these methods? Isn't this API breaking and tangential to the point of this PR?

It'll help if you read my review comments as well....